test/arrays.jl

```diff
@@ -44,4 +44,17 @@
     d = deepdiff(a1, [2, 4])
     @test removed(d) == [1, 3]
     @test added(d) == []
+
+    d = deepdiff([NaN], [NaN])
+    @test removed(d) == added(d) == []
+
+    d = deepdiff([missing], [missing])
+    @test removed(d) == added(d) == []
+
+    d = deepdiff([NaN], [])
+    @test d == d
```

Comment:
what is this `d == d` test testing? If this isn't a typo maybe add a comment explaining what's happening.

Reply:
It's testing equality on the `::DeepDiff` value itself. e.g. https://github.com/ssfrr/DeepDiffs.jl/blob/f21e03c47a8efae15713626c782ff0c3d4b87438/src/arrays.jl#L14 , which relies on `fieldequal`, which was also corrected in this PR.

Before the PR, that test would fail (for the same reason `NaN == NaN`is false).

Should I say that it tests `fieldequal`?